### PR TITLE
Ripgrep support

### PR DESCRIPTION
**Pull request by bartol**
Use ripgrep (if available) for faster code search.

### CODE REVIEW

File: vgrep.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ type cliArgs struct {
 	Debug       bool   `short:"d" long:"debug" description:"Verbose debug logging"`
 	Interactive bool   `long:"interactive" description:"Enter interactive shell"`
 	NoGit       bool   `long:"no-git" description:"Use grep instead of git-grep"`
+	NoRipgrep   bool   `long:"no-ripgrep" description:"Don't use ripgrep"`
```
Comment (vrothberg) [inline vgrep.go:36]: To be consistent with the NoHeader wording:
```suggestion
	NoRipgrep   bool   `long:"no-ripgrep" description:"Do not use ripgrep"`
```

File: vgrep.go [hunk 2]
```diff
@@ -170,6 +171,19 @@ func (v *vgrep) insideGitTree() bool {
 	return inside
 }
 
+// ripgrepInstalled returns true if ripgrep is installed
+func (v *vgrep) ripgrepInstalled() bool {
+	out, _ := exec.LookPath("rg")
```
Comment (vrothberg) [inline vgrep.go:176]: Please log the error. It can come in handy for debugging.

**Comment by vrothberg** [comment]
>  Would you extend the .install.tools make target to install ripgrep from https://github.com/BurntSushi/ripgrep/releases? 

You can install it to `./test/bin` and extend `$PATH` to `./test/bin` so we're not changing the host.

**Comment by bartol** [comment]
> Would you extend the `.install.tools` make target to install ripgrep from [BurntSushi/ripgrep/releases](https://github.com/BurntSushi/ripgrep/releases)?

Should I just replicate build steps from [Ripgrep's README](https://github.com/BurntSushi/ripgrep#building) in `./test/bin`?

**Comment by vrothberg** [comment]
> Should I just replicate build steps from [Ripgrep's README](https://github.com/BurntSushi/ripgrep#building) in `./test/bin`?

I think we can do a shortcut and download and extract the pre-built Linux binaries (see ripgrep-12.0.1-x86_64-unknown-linux-musl.tar.gz on the release page).

**Comment by bartol** [comment]
Did it. It's a little bit clunky but I couldn't find cleaner way.

**Comment by vrothberg** [comment]
Thanks!

```
./hack/tree_status.sh
tree is dirty, please commit all changes and sync the vendor.conf
```
We need to add `test/bin` to `.gitignore`. The vendor checks make sure that the go dependencies are up to date.

**Comment by bartol** [comment]
Now outputs:
```
$ make .install.tools
...
$ ./hack/tree_status.sh
tree is clean
```

**Comment by vrothberg** [comment]
https://github.com/vrothberg/vgrep/pull/65/checks?check_run_id=601591714 shows a build error caused by a recent commit.

**Comment by bartol** [comment]
Build is fixed but tests still fail because of output format.

**Comment by vrothberg** [comment]
Those need to be fixed in this PR. Do you need help?

**Comment by bartol** [comment]
I managed to fix errors with parsing ripgrep's output. All tests should now pass.

**Comment by bartol** [comment]
Should I add test with both `--no-git` and `--no-ripgrep`?

**Comment by vrothberg** [comment]
> Should I add test with both --no-git and --no-ripgrep?

Good idea, thanks!

**Comment by bartol** [comment]
Added test and match highlighting should be working properly now.

**Comment by bartol** [comment]
Squashed all 10 commits.

**Pull request merged**